### PR TITLE
Support cast (from boolen to decimal)

### PR DESCRIPTION
**Pull request by rui-mo**
Reuses `applyIntToDecimalCastKernel` to support cast (from boolean to decimal).

### CODE REVIEW

**Comment by rui-mo** [comment]
@mbasmanova Could you help review when you are free? Thanks.

File: velox/docs/functions/presto/conversion.rst [hunk 1]
```diff
@@ -652,6 +652,16 @@ Valid examples
 Cast to Decimal
 ---------------
 
+From boolean type
```
Comment (mbasmanova) [inline velox/docs/functions/presto/conversion.rst:655]: Perhaps, add a sentence to clarify that true value is converted to 1 and false to 0.

File: velox/expression/tests/CastExprTest.cpp [hunk 2]
```diff
@@ -1700,6 +1700,30 @@ TEST_F(CastExprTest, integerToDecimal) {
   testIntToDecimalCasts<int64_t>();
 }
 
+TEST_F(CastExprTest, boolToDecimal) {
+  // Bool to short decimal.
+  auto input =
+      makeFlatVector<bool>({true, false, false, true, true, true, false});
+  testComplexCast(
```
Comment (mbasmanova) [inline velox/expression/tests/CastExprTest.cpp:1707]: Curious why is decimal a complex type?

**Review by mbasmanova — APPROVED** [review]
Looks great % a couple of questions.

File: velox/expression/tests/CastExprTest.cpp (hunk 2 above)
Comment (rui-mo) [reply velox/expression/tests/CastExprTest.cpp:1707]: Seems `testComplexCast` is not restricted to complex type, but covers more cases than `testCast`, including constant input and dictionary input.

File: velox/docs/functions/presto/conversion.rst (hunk 1 above)
Comment (rui-mo) [reply velox/docs/functions/presto/conversion.rst:655]: Updated.

**Pull request closed without merging**